run.go

```diff
@@ -227,7 +227,12 @@ func runWithOptions(suiteName string, runner runner, opt Options) int {
 	runner.fmt = multiFmt.FormatterFunc(suiteName, output)
 
 	var err error
-	if runner.features, err = parser.ParseFeatures(opt.Tags, opt.Paths); err != nil {
+	if len(opt.FeatureContents) > 0 {
+		runner.features, err = parser.ParseFromBytes(opt.Tags, opt.FeatureContents)
+	} else {
+		runner.features, err = parser.ParseFeatures(opt.Tags, opt.Paths)
+	}
```

Comment:
Let's change logic here so that both `Paths` and bytes can work together. We can check for non-empty `opt.Paths` and append `runner.features`.

Reply:
hmm, I'm not so sure how simple this would be.  From what I can see, if someone wants to run through FeatureContents only, and they leave the Paths option blank, the current behavior will force them to use a features directory by default.  Either they'll have their features already in a features directory, or they won't, in which case there could be a failure due to non-existent directory, no?  I think it's maybe just simpler to say, either you can use feature contents, or features, but not both.  If someone is already parsing files and passing them in, I can't imagine it's much more challenging to also include from all features as opposed to from only a few.  what do you think?

Reply:
Hm, this is interesting point. 

> If someone is already parsing files and passing them in, I can't imagine it's much more challenging to also include from all features as opposed to from only a few.

Another case when this feature could be useful would be sourcing gherkin files from a remote storage (e.g. via HTTP) or generating/templating them by code. In such situation implementing additional file preloader might be a burden.

If we disable that default in case of feature contents presence it may be perceived as unexpected behavior.
```go
	if len(opt.Paths) == 0 && len(opt.Features) == 0 {
		inf, err := os.Stat("features")
		if err == nil && inf.IsDir() {
			opt.Paths = []string{"features"}
		}
	}
```

If we ignore `Paths` (make feature contents and paths mutually exclusive) this can also be perceived as unexpected behavior (that needs to be documented or maybe even have a check and explicit failure to not mislead user).
And implementing preloader can be a minor annoyance for a user in case they want both types of features.

From these options I'm leaning to disabling the default as it seems less disruptive overall.

Reply:
hmm, ok, then how about something like this?
```
	if len(opt.Paths) == 0 && len(opt.FeatureContents) == 0 {
		inf, err := os.Stat("features")
		if err == nil && inf.IsDir() {
			opt.Paths = []string{"features"}
		}
	}

	if len(opt.FeatureContents) > 0 {
		features, err := parser.ParseFromBytes(opt.Tags, opt.FeatureContents)
		if err != nil {
			fmt.Fprintln(os.Stderr, err)
			return exitOptionError
		}
		runner.features = append(runner.features, features...)
	}

	if len(opt.Paths) > 0 {
		features, err := parser.ParseFeatures(opt.Tags, opt.Paths)
		if err != nil {
			fmt.Fprintln(os.Stderr, err)
			return exitOptionError
		}
		runner.features = append(runner.features, features...)
	}
```

Reply:
Sounds good!